src/carve/main.clj

```diff
@@ -43,12 +43,18 @@
   (when-not (every? valid-path? paths)
     (throw (ex-info "Path not found" {:paths paths}))))
 
-(defn -main [& [flag opts & _args]]
+(defn main
+  [& [flag opts & _args]]
   (when-not (= "--opts" flag)
     (throw (ex-info (str "Unrecognized option: " flag) {:flag flag})))
   (let [opts (edn/read-string opts)
         _ (validate-opts! opts)
         report (impl/run! opts)]
     (when-let [r (:report opts)]
```

Comment:
I guess `main` can potentially return nil. I should it should return `0` instead of `nil` as the default exit code.

Reply:
Well I just wanted to change it for the report mode, I could change the rest as well and default to 0 but I think it might be wrong in way.
Or at least we would have to decide what to return in the cases where it's not a report, returning always `0` might not be correct imho, `nil` just says that it wasn't a report at least.

Reply:
What does `(System/exit nil)` do?

Reply:
Ah good question, it blows up 😄 
I only tried report otherwise I would have noticed.
Yeah well I can just add a `some->` maybe if that's ok, otherwise I can also just default to 0 what do you prefer @borkdude ?

Reply:
It's the same thing I guess since if we don't call System/exit directly the exit status code is 0 anyway.

Reply:
I think it's cleaner to have `main` always have it return a number, either 1 or 0. Just use `(or (...) 0)` I'd say.

Reply:
Ok cool